lib/counter_culture/counter.rb

```diff
@@ -49,78 +51,79 @@ def change_counter_cache(obj, options)
       # allow overwriting of foreign key value by the caller
       id_to_change = foreign_key_values.call(id_to_change) if foreign_key_values
 
-      if id_to_change && change_counter_column
-        delta_magnitude = if delta_column
-                            (options[:was] ? attribute_was(obj, delta_column) : obj.public_send(delta_column)) || 0
-                          else
-                            counter_delta_magnitude_for(obj)
-                          end
-        # increment or decrement?
-        operator = options[:increment] ? '+' : '-'
-
-        klass = relation_klass(relation, source: obj, was: options[:was])
-
-        # MySQL throws an ambiguous column error if any joins are present and we don't include the
-        # table name. We isolate this change to MySQL because sqlite has the opposite behavior and
-        # throws an exception if the table name is present after UPDATE.
-        quoted_column = if klass.connection.adapter_name == 'Mysql2'
-                          "#{klass.quoted_table_name}.#{model.connection.quote_column_name(change_counter_column)}"
+      return unless conditions_allow_change?(obj)
+      return unless id_to_change && change_counter_column
```

Comment:
lots of noise here but I the only change is switching to a guard clause since there are two now